src/dataset_processor.py

```diff
@@ -205,10 +205,43 @@ def load_assigned_reads(save_file_name, gffutils_db, multimapped_chr_dict):
         yield current_gene_info, read_storage
 
 
+def construct_models_in_parallel(sample, chr_id, dump_filename, args, multimapped_reads, current_chr_record,
+                                 io_support):
+    processor = DatasetProcessor(args, parallel=True)
+    processor.create_aggregators(sample)
+    gffutils_db = gffutils.FeatureDB(args.genedb, keep_order=True)
```

Comment:
why not keep this within DatasetProcessor?

Reply:
I tried it - not very good for parallelization

Reply:
hm, looks like i need to learn more about parallelization in python